### PR TITLE
Repository Cleanup

### PR DESCRIPTION
**Pull request by ElectronicsArchiver**
[Badge Suggestions]: https://img.shields.io/badge/Ｒｅｐｏｓｉｔｏｒｙ_Ｓｕｇｇｅｓｔｉｏｎｓ-609926?style=for-the-badge&logoColor=white&logo=GitHub
[Badge Changes]: https://img.shields.io/badge/Ｃｈａｎｇｅｓ-D81939?style=for-the-badge&logoColor=white&logo=GitLFS
[Badge Ask]: https://img.shields.io/badge/Ｐｌｅａｓｅ_Ａｓｋ_Ｍｅ-73398D?style=for-the-badge&logoColor=white&logo=AskUbuntu

[Widgets]: https://github.com/MarkedDown/Disable/blob/main/Features/Widgets.md
[Tabs]: https://github.com/MarkedDown/Disable/blob/main/Features/Tabs.md

<br>
<br>

<div align = center>

[<kbd> <br> Please Check The **Preview** <br> </kbd>][Preview]

</div>
  
<br>
<br>

## ![Badge Suggestions]

  <br>

- ***[Disable Unused Widgets][Widgets]***:

  - `Packages`

  <br>

- ***[Disable Unused Tabs][Tabs]***:

  - `Projects`
 
  - `Wiki`

<br>
<br>

## ![Badge Changes]

- Separated installation / usage section from README

- General formatting / spacing

- Added license badge

- Added quicklinks

- Added links to named resources

- Formatted configs

- Refactored plugin config
  
  *Waiting for test workflow.*

- Removed redundant LICENSE `.txt` extension

- Removed Changelog.md as it is literally a copy
  of the Releases content, linked releases instead.

  I would love to format the releases descriptions,
  however I don't think that's possible as a normal
  contributor.

<br>
<br>

## ![Badge Ask]

- ***If you have any questions.***

- ***In case you do not agree with some of the changes made in this PR***, <br>
  *please first let me know about them, before disregarding it entirely so*<br>
  *we can possibly find a middle ground and have it not go to waste.*

<br>
<br>

[Preview]: [#](https://github.com/ElectronicsArchiver/prettier-config-atomic)

### CODE REVIEW

File: prettier.config.js [hunk 1]
```diff
@@ -1,55 +1,93 @@
...
-    {
-      files: "{*.md}",
-      options: {
-        parser: "markdown",
-        proseWrap: "preserve",
-      },
-    },
-  ],
+    
+    singleQuote : false ,
```
Comment (aminya) [inline prettier.config.js:85]: Please revert the order of the properties because they were in the order of importance.

File: prettier.config.js [hunk 2]
```diff
@@ -1,55 +1,93 @@
...
+    files : '{*.md}',
+    options : {
+        proseWrap : 'preserve',
+        parser : 'markdown'
+    }
+}]
+
+
 module.exports = {
-  plugins: [require("prettier-plugin-jsdoc"), prettier_plugin_packagejson()],
```
Comment (aminya) [inline prettier.config.js:33]: Please inline the plugin property. The reason I wrote this in this format is that prettier.config.js can be a JSON file. Because of the bug in prettier with pnpm, I had to use a JS file. I don't want this config file to be primitive. It should be as declarative as possible.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,148 +0,0 @@
-## [3.0.10](https://github.com/atom-community/prettier-config-atomic/compare/v3.0.9...v3.0.10) (2022-06-22)
```
Comment (aminya) [inline CHANGELOG.md:1]: Why did you remove the changelog file? Please revert this.

File: Documentation/Installation.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+<br>
+<br>
+
+## Installing
+
+```shell
+npm install     \
+    --save-dev  \
+    prettier-config-atomic
```
Comment (aminya) [inline Documentation/Installation.md:39]: Make it one lineer. `\` implies bash

File: Documentation/Installation.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+<br>
+
+### Manual
+
+Alternatively you can also install `prettier` <br>
+in your `devDependencies` by yourself.
+
+*If you are using `npm`, the prettier* <br>
+*dependency is hoisted automatically.*
```
Comment (aminya) [inline Documentation/Installation.md:29]: Do not break the sentence. It should have a line length of 120

File: Documentation/Usage.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
+
+# Usage
+
+<br>
+
+## Configuration
+
+Add the following to your `package.json` :
+
+<br>
```
Comment (aminya) [inline Documentation/Usage.md:10]: I prefer to use simple markdown instead of HTML in markdown.

**Review by aminya — CHANGES_REQUESTED** [review]

File: prettier.config.js [hunk 7]
```diff
@@ -1,55 +1,93 @@
...
+    
+    
+    //  Static
+    
+    const isPackageJSON = /(^|\\|\/)package\.json$/;
+
+    const
+        parser_babylon = 'prettier/parser-babylon' ,
+        parser_babel = 'prettier/parser-babel' ,
+        package_sort = 'sort-package-json' ;
```
Comment (aminya) [inline prettier.config.js:25]: Break this into separate declaration statements. Eslint should have warned you about this. 
`;` will be removed once you run pnpm format.

**Review by aminya — CHANGES_REQUESTED** [review]

File: prettier.config.js [hunk 8]
```diff
@@ -1,55 +1,93 @@
...
+    
+    function preprocess(text,options){
+        
+        if(parser.preprocess)
+            text = parser.preprocess(text,options);
+
+        const { filepath } = options;
+        
+        if(filePath && isPackageJSON.test(filepath))
+            return sort(text);
```
Comment (aminya) [inline prettier.config.js:49]: Missing `{}` after the if statement. Eslint should have warned you

File: Documentation/Installation.md (hunk 5 above)
Comment (ElectronicsArchiver) [reply Documentation/Installation.md:29]: Why would you enforce 120 when 60 is the human readable limit?

File: Documentation/Usage.md (hunk 6 above)
Comment (ElectronicsArchiver) [reply Documentation/Usage.md:10]: Markdown is an extension to HMTL not the other way around.

File: prettier.config.js [hunk 9]
```diff
@@ -1,55 +1,93 @@
...
+    
+    //  Static
+    
+    const isPackageJSON = /(^|\\|\/)package\.json$/;
+
+    const
+        parser_babylon = 'prettier/parser-babylon' ,
+        parser_babel = 'prettier/parser-babel' ,
+        package_sort = 'sort-package-json' ;
+
```
[the remark this answers is not in the document]
Comment (ElectronicsArchiver) [reply prettier.config.js:26]:
```js
const parser_babylon = 'prettier/parser-babylon';
cosnt parser_babel = 'prettier/parser-babel';
const package_sort = 'sort-package-json';
```

You call this more readable?

File: Documentation/Installation.md (hunk 4 above)
Comment (autumnblazey) [reply Documentation/Installation.md:39]: I think it means to just write `npm install --save-dev prettier-config-atomic`, all on one line, no need to break it into seperate lines with `\`

File: Documentation/Installation.md (hunk 5 above)
Comment (aminya) [reply Documentation/Installation.md:29]: For me, 120 is more readable and easier to work with than <80. Also, almost all the reading environments have soft wrapping enabled. You can enforce your desired soft wrap in your reading environment (instead of the writing source).

File: CHANGELOG.md (hunk 3 above)
Comment (aminya) [reply CHANGELOG.md:1]: Changelog files are shown inside Atom, but the links are on GitHub only.

File: prettier.config.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (autumnblazey) [reply prettier.config.js:26]: I see what you mean in that it could be called less readable, in that there is more `const` that adds to clutter. But to me at least, it is clearer what each lines' intent is with the `const` keyword in the beginning of each line clearly saying that it is a const declaration. Also each variable name is still aligned with the others, so theres no problem there

File: Documentation/Usage.md (hunk 6 above)
Comment (aminya) [reply Documentation/Usage.md:10]: `<br>` is not required once we have `enter` in Markdown.

File: Documentation/Installation.md (hunk 4 above)
Comment (autumnblazey) [reply Documentation/Installation.md:39]: yeah, but because it is a pretty short command I think it is nicer to have it in one line. If there were multiple packages being installed, then I think I would agree that it makes more sense to break it up into separate lines

File: prettier.config.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (ElectronicsArchiver) [reply prettier.config.js:26]: @aminya The only thing it warned me about is a non-existant `)` in the `.eslintrc` file.

File: Documentation/Installation.md (hunk 4 above)
Comment (aminya) [reply Documentation/Installation.md:39]: I use Powershell and nushell which don't support `\`. 

> more readable

This is subjective. It is not more readable for me as an example.
